### DISCUSSION TITLE
Search for existing PR does not prefer existing PRs to the baseBranch

### DISCUSSION THREAD

**Discussion by kroonprins** [renovatebot/renovate#23929] [Suggest an Idea]
In the logic for creating a PR, renovate searches for an existing PR using `platform.getBranchPr(branchName)`. For the azure platform, if the repository contains multiple PRs for that branch to different target branches, then it will return the first branch it encounters: https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/index.ts#L333C25-L333C25. Would it be feasible or make sense to change this so that it first favors a PR to the expected base branch, if it exists? And only falls back on `prsFiltered[0]` if it doesn't exist?

It is perhaps a bit of a strange situation though to have multiple PRs for a branch to different target branches, but it is a situation I have (for a gitops monorepo with a branch-per-environment containing kubernetes resources and trying to update image tags with the "kubernetes" renovate manager). Currently it happens that renovate selects a PR for another target branch and then fails with an error from the azure devops api `TF401179: An active pull request for the source and target branch already exists` because it tries to update the selected PR with a new target branch and there is already another PR with that target branch.

**Comment by rarkins** [comment]
Renovate by default uses different branch names if you have multiple baseBranches, so this problem normally shouldn't exist unless you have been changing your default repo branch

**Reply by kroonprins** [reply]
Unless I misunderstand what you meant, it's not really a case of multiple baseBranches for my situation. The situation is that I have in the renovate configuration only one baseBranch. Renovate creates a renovate branch and PR for this baseBranch, but afterwards there is another process that can create other PRs for that same renovate branch towards other target branches. When setting this up, initially what I tried to achieve was that renovate creates just the branch, but that my other process takes care of creating all PRs where needed, but it didn't seem possible to prevent renovate from creating a PR. It also didn't seem possible to have renovate create multiple PRs with other targets than the configured baseBranch. So that is why I currently have the situation where renovate creates the PR for the baseBranch and another process for creating other PRs for that branch. It's these other PRs created by this other process that are getting sometimes picked up by this `platform.getBranchPR(branchName)`.

**Reply by rarkins** [reply]
I didn't understand that you were creating multiple PRs. You are welcome to submit a PR improving Renovate as you describe

**Reply by kroonprins** [reply]
Thanks, I have created a PR for it
